### ISSUE TITLE
Fix Docsify issues or investigate alternatives like mdbook

### ISSUE THREAD

**Issue by dae** [ankitects/anki#977]
Docsify has an annoying bug where sidebar clicks will not scroll:

https://anki.tenderapp.com/discussions/beta-testing/1852-anki-docs-search-button-anchors-are-not-working-ie-jumping

Even if we're able to fix that issue, links to subsections near the end of a page aren't ideal - since the page can't scroll past the bottom, the section that we want the user to read may appear near the bottom of the window, requiring the user to skip their eyes over a number of sections above. To address that we'd need to pad out the bottom of each section as well.

It may be worth considering alternatives like mdbook instead. Mdbook positives:

- can create a print.html file containing the entire manual, which some users prefer over individual pages
- supports user-selectable themes (dark mode)

Downsides:

- unable to generate the sidebar automatically - sections need to be manually listed out. Subsections can be automatically generated with an add-on, but they get added to the top of the right hand side, not to the sidebar. This may require a fair amount of work to adjust to.
- No support for a language selection button. We don't take advantage of this in Docsify at the moment, but it would be nice if in the future users could switch between available translations on a per-section basis.
- Not quite as easy to deploy as Docsify

**Comment by dae** [comment]
I migrated [AnkiMobile's manual](https://docs.ankimobile.net/) over to mdbook a few days ago. The links to separate sections had to be manually updated which was a pain - while there's a link checker available for external links, it doesn't understand #anchor links. Happy with the faster page load speeds and the fact that the interface responds nicely on phones (and even follows the set dark mode). There were some out-of-date sections that needed updating at the same time, so it took longer than expected.

While there is a TOC generator that can embed a TOC at the top of right section, I opted for unrolling the nested sections into separate top-level sections for AnkiMobile. Anki's manual is larger and that probably won't be practical, so if migrating it, we may need to either put the TOC on the top right of each section, or start splitting the files up and linking subsections in the left outline. The former is much easier than the latter, so would probably be the best way to start, as we could gradually migrate to the latter if appropriate.

We're currently using https://github.com/peaceiris/actions-mdbook to build the pages, but that does not appear to support anything except the base mdbook, so we'd need to write our own action, or I'd need to set up Buildkite to build the manual with  mdbook-toc installed.
